### PR TITLE
feat: mysql connector

### PR DESCRIPTION
**Pull request by arashsheyda**
### 🔗 Linked issue

related to #32

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

this PR adds support for `mysql` connector using `mysql2` + an example of `mysql` (also I'm sorry in advanced if I over stepped, I was checking this fantastic module and couldn't resist trying)

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/connectors/mysql2.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import type { Connector, Statement } from "../types";
+
+export default function mysqlConnector(opts: mysql.ConnectionOptions) {
+  let _connection: mysql.Connection | undefined;
+  const getConnection = async () => {
+    if (_connection) {
+      return _connection;
+    }
+
+    _connection = await mysql.createConnection({
```
Comment (arashsheyda) [inline src/connectors/mysql2.ts:12]: I wan't sure if we want to have some `options` as default, for example we can set the `host` to `localhost` or `password` to `""`...

Comment (rafazafar) [reply src/connectors/mysql2.ts:12]: Would be nice to default to root@localhost:3306 but unsure how useful in practice :)

Comment (arashsheyda) [reply src/connectors/mysql2.ts:12]: yes, or we can set `root@localhost:3306` to default only in dev

**Review by pi0 — APPROVED** [review]
Thanks ❤️

**Comment by sys-tps** [comment]
Thank you very much !
I wait for this a very long time...

**Pull request merged**